tests/test_scheduler.py

```diff
@@ -59,97 +66,154 @@ def test_simple_interval_scheduler(start_scheduler, start_cli):
 
 
 @pytest.mark.skipif(os.getenv("CI") == "true", reason="test skipped on CI")
```

Comment:
I think you can safely remove all these skip on CI as test should work now with your changes